src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -2142,32 +2195,46 @@ private static List<CompletionResult> GetResultForAttributeArgument(CompletionCo
             Type attributeType = null;
             string argName = string.Empty;
             Ast argAst = completionContext.RelatedAsts.Find(static ast => ast is NamedAttributeArgumentAst);
-            NamedAttributeArgumentAst namedArgAst = argAst as NamedAttributeArgumentAst;
-            if (argAst != null && namedArgAst != null)
+            AttributeAst attAst;
+            if (argAst is NamedAttributeArgumentAst namedArgAst)
             {
-                attributeType = ((AttributeAst)namedArgAst.Parent).TypeName.GetReflectionAttributeType();
+                attAst = (AttributeAst)namedArgAst.Parent;
+                attributeType = attAst.TypeName.GetReflectionAttributeType();
                 argName = namedArgAst.ArgumentName;
                 replacementIndex = namedArgAst.Extent.StartOffset;
                 replacementLength = argName.Length;
             }
             else
             {
                 Ast astAtt = completionContext.RelatedAsts.Find(static ast => ast is AttributeAst);
-                AttributeAst attAst = astAtt as AttributeAst;
-                if (astAtt != null && attAst != null)
+                attAst = astAtt as AttributeAst;
+                if (attAst is not null)
```

Comment:
```suggestion
                if (astAtt is AttributeAst attAst)
```

Reply:
This won't work. attAst is used outside that if statement so I can't declare it there.

Reply:
I see. Then ignore this comment.